### PR TITLE
🧱component: table component , fixes #29

### LINKED ISSUE #29: Products table page component
- Create a page component that will be used in products page route `records/products`.
- Create the table component apart with pagination.
- In the root of the page as a server component fetch  the products using `productsService.listProducts` then pass them to `ProductsPage` component as a prop which is already created.

### PR DESCRIPTION
**Pull request by kaufon**

### CODE REVIEW

File: apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts [hunk 1]
```diff
@@ -1,40 +1,211 @@
...
 
-export const useProductsTableHook = (page: number) => {
-  const useapi = useApi()
+export const useProductsTableHook = () => {
+  const [pageState, setPage] = useQueryState('page', parseAsInteger)
+  const page = pageState ?? 1
+  const [filterByNameValueState, setFilterByNameValue] = useQueryState('name')
+  const filterByNameValue = filterByNameValueState ?? ''
+
+  const useApiInstance = useApi()
```
Comment (JohnPetros) [inline apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts:13]: Always desestructure the object returned by `useApi` hook to get its services inside it.

File: apps/web/src/ui/components/commons/search-component/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+interface SearchComponentsProps {
+  onSearchChange: (value: string) => void
+  filterByNameValue: string
+}
+
+export const TableSearchComponent = ({
+  onSearchChange,
+  filterByNameValue,
+}: SearchComponentsProps) => {
+  return useMemo(() => {
```
Comment (JohnPetros) [inline apps/web/src/ui/components/commons/search-component/index.tsx:14]: `useMemo` is completely desnecessary here. this hook usually used to memoize heavy calculations only

File: apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts [hunk 3]
```diff
@@ -1,40 +1,211 @@
...
+  // Filter products by name Logic :)
+  const filteredItemsByName = useMemo(() => {
+    if (!filterByNameValue) return products
+    return products.filter((product) =>
+      product.name.toLowerCase().includes(filterByNameValue.toLowerCase()),
+    )
+  }, [products, filterByNameValue])
+
+  // Pagination logic :)
+  const rowsPerPage = 10
```
Comment (JohnPetros) [inline apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts:38]: Use the constant `PAGINATION` from core package instead

File: apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts [hunk 4]
```diff
@@ -1,40 +1,211 @@
...
+    filterByNameValue,
+    onSearchChange,
+    paginatedProducts: paginatedItems,
+    totalPages,
+    loading,
+  }
 }
+
+//this is a VERY bad mock data,but use here for tests while backend is even worst than this mock
+const generateMockProducts = () => [
```
Comment (JohnPetros) [inline apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts:66]: Use `ProductsFaker.fakeManyDto()` instead

File: apps/web/src/ui/components/commons/search-component/index.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { Input } from '@nextui-org/react'
+import { Search } from 'lucide-react'
+import { useState, useCallback, useMemo } from 'react'
+
+interface SearchComponentsProps {
+  onSearchChange: (value: string) => void
+  filterByNameValue: string
+}
+
+export const TableSearchComponent = ({
```
Comment (JohnPetros) [inline apps/web/src/ui/components/commons/search-component/index.tsx:10]: The name of each component must no receive 'component' as sufix.

**Review by JohnPetros — CHANGES_REQUESTED** [review]

File: apps/web/src/ui/components/commons/search-component/index.tsx (hunk 2 above)
Comment (kaufon) [reply apps/web/src/ui/components/commons/search-component/index.tsx:14]: there will be when an array of 1000+ products is being filtered letter by letter

**Pull request merged**